core/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -144,9 +162,13 @@ public Value parse(XContentParser parser, Value value, Context context) throws I
                 if (currentFieldName == null) {
                     throw new IllegalStateException("[" + name  + "] no field found");
                 }
-                assert fieldParser != null;
-                fieldParser.assertSupports(name, token, currentFieldName, context.getParseFieldMatcher());
-                parseSub(parser, fieldParser, currentFieldName, value, context);
+                if (fieldParser == null) {
+                    assert ignoreUnknownFields : "this should only be possible if configured to ignore known fields";
```

Comment:
is it possible that somebody doesn't provide the suppler but has ignoreUnkonwnFields set to false? in that case we wouldn't get here anyways right? just checking... :)